### DISCUSSION TITLE
Configure purely with RENOVATE_CONFIG environment variable fails.

### DISCUSSION THREAD

**Discussion by nielsbasjes** [renovatebot/renovate#9970] [Request Help] [answered]
I have configured renovate to run for my project in the internal Gitlab installation we have.
Reading https://docs.renovatebot.com/self-hosting/#configuration I understand that it should be possible to configure renovate using the environment variable RENOVATE_CONFIG.

So I have created an internal renovate-config project with the config files.
In the project that needs to be checked I put this in my .gitlabci.yaml 

```
include:
  - project: 'developertools/renovate-config'
    file: 'renovate.gitlab-ci.yml'
```
In this imported file I have (among a lot of other things)
```
variables:
    RENOVATE_EXTRA_FLAGS: ${CI_PROJECT_NAMESPACE}/${CI_PROJECT_NAME}
    RENOVATE_REQUIRE_CONFIG: 'true'
    RENOVATE_CONFIG: '{ "extends": [ "local>developertools/renovate-config" ] }'
```

When I run renovate with this it reports the project is disabled.
```
  script:
    - renovate $RENOVATE_EXTRA_FLAGS
```

Digging through the debug output it seemed to me it was explicitly looking for a config file and because that was missing it did nothing.

Summary of the debug output I found:
```
DEBUG: No config file found on disk - skipping
DEBUG: Detected config in env RENOVATE_CONFIG
       "config": {"extends": ["local>developertools/renovate-config"]}
```
Then later on the "Combined config" shows exactly the config I want.

Near the end of the debug output I get 
```
DEBUG: findFile(renovate.json) (repository=developertools/renovate-experiment)
DEBUG: Initializing git repository into ... 
DEBUG: git clone completed ...
       "durationMs": 533
DEBUG: latest repository commit ...
DEBUG: findFile(renovate.json5) (repository=developertools/renovate-experiment)
DEBUG: findFile(.github/renovate.json) (repository=developertools/renovate-experiment)
DEBUG: findFile(.github/renovate.json5) (repository=developertools/renovate-experiment)
DEBUG: findFile(.gitlab/renovate.json) (repository=developertools/renovate-experiment)
DEBUG: findFile(.gitlab/renovate.json5) (repository=developertools/renovate-experiment)
DEBUG: findFile(.renovaterc) (repository=developertools/renovate-experiment)
DEBUG: findFile(.renovaterc.json) (repository=developertools/renovate-experiment)
DEBUG: config file not found (repository=developertools/renovate-experiment)
 INFO: Repository is disabled - skipping (repository=developertools/renovate-experiment)
```
And though this test project has deliberatly some older dependencies nothing happens.

So now I have in my project a `renovate.json` with the exact same content as the above environment variable and now it does work as expected.

My question here is what am I doing wrong?
Did I misunderstand the documentation? 
Or is this a bug in renovate I should report?

**Comment by viceice** [comment]
if you don't want to have a `renovate.json` in your repo, you need to configure [`onboarding=false`](https://docs.renovatebot.com/self-hosted-configuration/#onboarding]

Assuming you also have configured env vars for gitlab platform, checkout our gitlab pipeline template here https://gitlab.com/renovate-bot/renovate-runner/-/blob/master/templates/_common.gitlab-ci.yml#L1

**Reply by nielsbasjes** [reply]
Yes, I already have all of this.

**Comment by nielsbasjes** [comment, marked as answer]
Found it.
I have downloaded the entire source of renovate and started digging.

I found one of the error messages from my debug log in this code fragment:
```
export const isOnboarded = async (config: RenovateConfig): Promise<boolean> => {
  logger.debug('isOnboarded()');
  const title = `Action required: Add a Renovate config`;
  // Repo is onboarded if admin is bypassing onboarding and does not require a
  // configuration file.
  if (config.requireConfig === false && config.onboarding === false) {
    // Return early and avoid checking for config files
    return true;
  }
  if (await configFileExists()) {
    await platform.ensureIssueClosing(title);
    return true;
  }
  logger.debug('config file not found');
```

So apparently `requireConfig` really means `requireConfigFile`.
The config was present (loaded and valid) yet it was not in a file making this check fail.

**Reply by viceice** [reply]
![image](https://user-images.githubusercontent.com/1798109/117978094-cbdd9d00-b331-11eb-876a-823ee79f3e6c.png)

This is pretty clear to me. @HonkingGoose Can you propose an improvement to this if it's not clear enough?

**Reply by rarkins** [reply]
I'll push a quick text update shortly

**Reply by nielsbasjes** [reply]
Please also put this in the (now empty) section in self-hosted-configuration.md

**Reply by rarkins** [reply]
That's where it is

**Reply by HonkingGoose** [reply]
PR #9973 fixes the documentation.
